addons/digest/tests/test_digest.py

```diff
@@ -0,0 +1,75 @@
+from unittest.mock import patch
+import random
+
+from dateutil.relativedelta import relativedelta
+from lxml import html
+
+from odoo.addons.mail.models.mail_mail import MailMail
+
+from odoo import fields
+from odoo.tests import TransactionCase
+
+
+class TestDigestStuff(TransactionCase):
+    def test_digest_numbers(self):
+        self._setup_messages()
+
+        d = self.env['digest.digest'].create({'name': "My Digest", 'kpi_mail_message_total': True})
+
+        # subscribe the demo user so at least one mail gets sent
+        demo = self.env.ref('base.user_demo')
+        d_demo = d.with_env(self.env(user=demo))
+        d_demo.action_subscribe()
+        self.assertTrue(
+            d_demo.is_subscribed,
+            "check the user was subscribed as action_subscribe will silently "
+            "ignore subs of non-employees"
+        )
+
+        # digest creates its mails in auto_delete mode so we need to capture
+        # the formatted body during the sending process
+        body = None
+        def send(self, *args, _original=MailMail.send, **kwargs):
+            nonlocal body
+            body = self.body_html
+            return _original(self, *args, **kwargs)
+
+        with patch.object(MailMail, 'send', autospec=True, wraps=MailMail, side_effect=send):
```

Comment:
Why not using existing mail tools and mocks ? I know there are maybe not the cleanest but I would prefer using it than having yet-another-mock-not-completely-like others.

Reply:
Because it didn't work when I tried and I've no idea which one is supposed to intercept message bodies? The one I tried would store the MailMessage objects, but since the mails are auto_delete by the time it returned the mails had been deleted and there was no way to inspect the body.